### DISCUSSION TITLE
Strange behavior of genre="..." filter

### DISCUSSION THREAD

**Discussion by stefan-i** [quodlibet/quodlibet#3946] [Q&A] [answered]
Hi,
I have a simple filter I want to use to separate my classical genre albums from all others:
_genre="classical"_
and
_genre!="classical"_

However, it does not work as expected. Whenever I filter with _genre="classical"_ (or for any other genre) the result list also contains all tracks from other genres that do not have an "album" tag and (even weirder) all albums that are called "Live", see e.g. here:
<img width="1917" alt="Screenshot_20220321_212745" src="https://user-images.githubusercontent.com/13986045/159358315-b415f2d9-dd00-4fb3-8358-1cdd88a4352e.png">
Conversely, these wrong files are then missing with the _genre!="classical"_ filter, where they should normally appear (because they do not have the classical genre tag).
This behavior is shown in "Album List" , "Album Collection" and "Cover Grid" browsers, but not in "Paned Browser" when I select genre manually in the genre tab.
Am I doing something wrong or is this a bug?
I am using Quodlibet 4.3.0 on MX Linux 21.
Thank you for your help!

**Comment by declension** [comment, marked as answer]
It's hard to say without a smaller example and seeing all tags for all those files... but, though unlikely, it sounds like each of those albums could have _at least one_ track with (at least) a `genre` value of _classical_. -- it's an _any_ aggregation for albums, not an _all_. 

Also, because of how albums themselves are identified (see [FAQ entry on merged albums](https://quodlibet.readthedocs.io/en/latest/guide/faq.html#i-have-two-albums-with-the-same-name-which-are-merged-in-the-album-list)) you might get additional weirdness here (my recommendation: tag with one of `labelid` or perhaps better, musicbrainz metadata - you can use the plugin)

**Reply by stefan-i** [reply]
Thank you for your reply. In fact there is _one_ classical album in my collection called "Live", so this means _any_ album with the same name from other genres will also be listed by the genre="classical" query? Also there are some tracks with classical genre but without album name, and this means all tracks without album name will be listed as "classical" as well? If yes, this would explain the result.

The problem is I have a very large collection so manually changing or adding stuff would be very time-consuming.
By the way, could the problem be caused by the fact that I do not use "artist" or "album artist" tags for classical works but only use "composer" instead. Would it solve the issue if I batch copy the contents of the "composer" tag also to "album artist"?

I was also wondering if I could circumvent the issue by adding some additional arguments with an AND operator. In my case that the _album title may not be empty_ and the _album title may not be "Live"_ . Is there a way to do this?

**Reply by declension** [reply]
> In fact there is _one_ classical album in my collection called "Live", so this means _any_ album with the same name from other genres will also be listed by the genre="classical" query? 

Yes this sounds like it then, it's a bit weird for sure, but this "reverse lookup" effect has come up a few times (can't find the issue numbers). The key here might be to put `labelid` (etc) or Musicbrainz album ID info on that _one_ album and see how that helps.

> Also there are some tracks with classical genre but without album name, and this means all tracks without album name will be listed as "classical" as well? If yes, this would explain the result.

I can't remember but it sounds possible. However, there's quite a lot of special-case handling for "Songs without an album". 
 
> The problem is I have a very large collection so manually changing or adding stuff would be very time-consuming. 

Sure. Though you can run, say, the Musicbrainz plugin on _many_ albums in one go (though still have to interact per-album). 
:information_source:  Finding them is easy too with `musicbrainz_albumid=''`


> By the way, could the problem be caused by the fact that I do not use "artist" or "album artist" tags for classical works but only use "composer" instead.  Would it solve the issue if I batch copy the contents of the "composer" tag also to "album artist"?

This is what I do, FWIW. Well, similar enough. There are so many things (QL and more) that want `artist` or `albumartist` to be populated that it seems hard to fight it. Also, because we have `~people`, it's actually handled pretty nicely.


> I was also wondering if I could circumvent the issue by adding some additional arguments with an AND operator. In my case that the _album title may not be empty_ and the _album title may not be "Live"_ . Is there a way to do this?

Yes easy. Check out the [search user guide](https://quodlibet.readthedocs.io/en/latest/guide/searching.html)

`&(album = /./, album != "Live")` might do it..

**Reply by stefan-i** [reply]
Thank you for your explanations. About your last suggestion, I tried the following as the complete filter:
&(genre="classical", album = /./, album != "Live")
It seems to work fine like this and all unwanted albums are removed.
